extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -46,37 +46,54 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 				.withAdditionalProperties(propertyBag).send();
 
 			// prompt user for connection profile to get connection info
-			quickPickStep = 'getConnectionInfo';
-			connectionInfo = await vscodeMssqlApi.promptForConnection(true);
-			if (!connectionInfo) {
-				// User cancelled
-				return;
-			}
-			TelemetryReporter.createActionEvent(TelemetryViews.CreateAzureFunctionWithSqlBinding, TelemetryActions.startCreateAzureFunctionWithSqlBinding)
-				.withAdditionalProperties(propertyBag).withConnectionInfo(connectionInfo).send();
-
-			// list databases based on connection profile selected
-			connectionURI = await vscodeMssqlApi.connect(connectionInfo);
-			listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
-			const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
-				canPickMany: false,
-				title: constants.selectDatabase,
-				ignoreFocusOut: true
-			}));
+			while (true) {
+				connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					// User cancelled
+					return;
+				}
+				quickPickStep = 'getConnectionInfo';
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
```

Comment:
What would be even better IMO would be to have the quickpick prompt to select a connection also have the option to test a connection - which could then have the progress notification on the quickpick itself. 

Having the user have to jump around where their attention is during the flow isn't great

Reply:
This is what we are doing however in this flow. Our focus is the connected scenario. 

The flow seems reasonable to me: 
1. User chooses a connection profile 
2. Progress report shown while connecting
3. Error shown (mssql connection error) if unable to connect
4. User is shown connection profile list again

Reply:
I don't understand how your comment relates to mine. The flow would stay the same - I'm suggesting that the progress notification be done on the quickpick itself. 

See [QuickInput.busy](https://github.com/microsoft/azuredatastudio/blob/2bc6a0cd012de69504061a876070cfa43533fedc/src/vs/vscode.d.ts#L9926)

(This would of course need to be done in the vscode mssql extension though)

Reply:
So instead I should pass in busy (optional) parameter to promptForConnection and then use that to create the progress notification on the vscode-mssql side?

Reply:
After looking a bit at the QuickInput.busy and how promptForConnection is setup we call `promptItemChoice` here - https://github.com/microsoft/vscode-mssql/blob/main/src/views/connectionUI.ts#L90
This takes in options of type `vscode.QuickPickOptions` and `busy` is not one of the properties on the list.

Reply:
Yes, you'll have to modify it to create our own QuickPick and then you'll have full control over it so can set properties like busy. 

Given the amount of refactoring that would require though I guess we can do that later. But please make an issue to track doing that.

Reply:
Created Issue here: https://github.com/microsoft/vscode-mssql/issues/17325